kernel/src/engine/arrow_expression/mod.rs

```diff
@@ -213,6 +213,24 @@ impl Scalar {
     }
 }
 
+impl ArrayData {
+    /// Convert kernel [`ArrayData`] to Arrow [`ArrayRef`].
+    pub fn to_arrow(&self) -> DeltaResult<ArrayRef> {
+        let arrow_data_type = ArrowDataType::try_from_kernel(self.array_type().element_type())?;
+
+        #[allow(deprecated)]
+        let num_elements = self.array_elements().len();
```

Comment:
Do we really need to deprecate `ArrayData.array_elements()`? I rely on being able to access the values in `ArrayData`, so I would need a suitable replacement when we decide to remove it.

Reply:
Yeah agree. @lbhm can you make an issue to un-deprecate and reference this comment chain, thanks!